pkg/provision/engine.go

```diff
@@ -867,10 +868,38 @@ func (e *NativeEngine) uninstallDeployment(ctx context.Context, dl *gridtypes.De
 
 }
 
+func getZmountSize(wl *gridtypes.WorkloadWithID) (gridtypes.Unit, error) {
+	dataI, err := wl.WorkloadData()
+	if err != nil {
+		return 0, err
+	}
+	data, ok := dataI.(*zos.ZMount)
+	if !ok {
+		return 0, errors.New("not a zmount")
+	}
+	return data.Size, nil
+}
+
 func (e *NativeEngine) installDeployment(ctx context.Context, getter gridtypes.WorkloadGetter) {
 	for _, typ := range e.order {
 		workloads := getter.ByType(typ)
 
+		if typ == zos.ZMountType {
+			sort.Slice(workloads, func(i, j int) bool {
+				sizeI, err := getZmountSize(workloads[i])
+				if err != nil {
+					log.Error().Err(err).Stringer("id", workloads[i].ID).Msg("failed to get zmount size")
+				}
+
+				sizeJ, err := getZmountSize(workloads[j])
+				if err != nil {
+					log.Error().Err(err).Stringer("id", workloads[j].ID).Msg("failed to get zmount size")
+				}
+
+				return sizeI > sizeJ
```

Comment:
How can we depend on sizeI and sizeJ while they're supposed to be the zero value returns of getZmountSize in case of errors? I believe each branch after the error check should return a bool (e.g false in the first case and true in the 2nd one)

Reply:
Also, maybe extract the code t sort ZMount workloads into its own function, so it could be easily testable